packages/common/src/FindContractVersion.js

```diff
@@ -0,0 +1,48 @@
+const assert = require("assert");
+
+/**
+ * Get the version and type of a financial contract deployed using the official UMA contract factories.
+ * Note: all inputs and outputs are expressed as fixed-point (scaled by 1e18) BNs.
+ * @param {Object} web3 instance
+ * @param {string} contractAddress address of the contract in question
+ * @return {Object} contract name & version
+ */
+async function findContractVersion(contractAddress, web3) {
+  assert(web3, "Web3 object must be provided");
+  assert(contractAddress, "Contract Address Must be provided");
+  const contractCode = await web3.eth.getCode(contractAddress);
+  const contractCodeHash = web3.utils.soliditySha3(contractCode);
+  return versionMap[contractCodeHash] || {};
+}
+
+const versionMap = {
```

Comment:
This seems like the cleanest way to detect version, +1. Another idea would have been detecting presence of certain methods that exist in one version but not another, but that seems overly complex.

Reply:
ye, I messed around with that & it worked relatively well. However, I felt that this would be more consistent as there is not always going to be a 1:1 mapping between some methods being present and a contract version. for example, how do you tell the difference between 1.2.1 and 1.2.2 if they have the same methods?

Reply:
Yeah you'd have to test logic of the functions as well which sounds like too much code